### PR TITLE
added tests for batch operations

### PR DESCRIPTION
**Pull request by IvanAdamovNeon**

### CODE REVIEW

File: integration/tests/basic/test_batch_operations.py [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+            request: pytest.FixtureRequest,
+            common_contract,
+            json_rpc_client: JsonRPCSession,
+    ):
+        """
+        sends a batch operation request with 100 calls to the same function with different parameters
+        """
+        account = self.accounts[0]
+        evm_version: Version = self.web3_client.get_evm_version_number()
+        data_or_input_field_name = "data" if evm_version < Version("1.13.0") else "input"
```
Comment (kristinaNikolaevaa) [inline integration/tests/basic/test_batch_operations.py]: Can you please explain why this condition is here?

File: integration/tests/basic/test_batch_operations.py [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+        results = response.json()
+        msg = f"Sent batch length {len(batch)}, but received {len(results)} results"
+        assert len(results) == len(batch), msg
+
+        for result in results:
+            assert "error" not in result
+            assert rpc_checks.is_hex(result["result"])
+
+    def test_batch_operations_negative(
```
Comment (kristinaNikolaevaa) [inline integration/tests/basic/test_batch_operations.py]: I suggest to add one case with all positive and 1 negative requests somewhere in the middle
And some request wich return error from a contract ( not bad parameters error)

File: integration/tests/basic/test_batch_operations.py [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+            self,
+            request: pytest.FixtureRequest,
+            common_contract,
+            json_rpc_client: JsonRPCSession,
```
Comment (kristinaNikolaevaa) [inline integration/tests/basic/test_batch_operations.py]: looks like you don't use json_rpc_client here and common_contract

File: integration/tests/basic/test_batch_operations.py [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+from web3 import Web3
+
+from integration.tests.basic.helpers import rpc_checks
+from utils.accounts import EthAccounts
+from utils.apiclient import JsonRPCSession
+from utils.web3client import NeonChainWeb3Client
+
+
+@pytest.mark.usefixtures("accounts", "web3_client")
+class TestBatchOperations:
```
Comment (kristinaNikolaevaa) [inline integration/tests/basic/test_batch_operations.py]: let's move these tests to rpc folder

File: integration/tests/basic/test_batch_operations.py [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+    web3_client: NeonChainWeb3Client
+
+    def test_batch_operations_same_function(
+            self,
+            request: pytest.FixtureRequest,
+            common_contract,
+            json_rpc_client: JsonRPCSession,
+    ):
+        """
+        sends a batch operation request with 100 calls to the same function with different parameters
```
Comment (kristinaNikolaevaa) [inline integration/tests/basic/test_batch_operations.py]: please add descriptions to md files

File: integration/tests/basic/test_batch_operations.py [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+        """
+        account = self.accounts[0]
+        evm_version: Version = self.web3_client.get_evm_version_number()
+        data_or_input_field_name = "data" if evm_version < Version("1.13.0") else "input"
+
+        batch = []
+        expected_results = []
+
+        for id_ in range(1, 101):
```
Comment (kristinaNikolaevaa) [inline integration/tests/basic/test_batch_operations.py]: how long it takes to run this test?Is it still fast if there 100 requests? Just curious

File: integration/tests/basic/test_batch_operations.py (hunk 1 above)
Comment (IvanAdamovNeon) [reply integration/tests/basic/test_batch_operations.py]: the up-to-date documentation states that eth_call params expect field "input"
https://ethereum.org/en/developers/docs/apis/json-rpc/#eth_call

but it turned out that with older evm versions, the field has to be "data"
https://github.com/ethereum/go-ethereum/issues/28608#issue-2010585064

File: integration/tests/basic/test_batch_operations.py (hunk 6 above)
Comment (IvanAdamovNeon) [reply integration/tests/basic/test_batch_operations.py]: 2.5 sec excluding setup

File: integration/tests/basic/test_batch_operations.py (hunk 1 above)
Comment (kristinaNikolaevaa) [reply integration/tests/basic/test_batch_operations.py]: but it's not our evm version 
it's go ethereum
You check version or neon evm

You can check here only current behavior

**Review by kristinaNikolaevaa — APPROVED** [review]

**Pull request merged**